driver-core/src/main/com/mongodb/internal/async/client/AsyncAggregateIterableImpl.java

```diff
@@ -157,6 +157,9 @@ public void toCollection(final SingleResultCallback<Void> callback) {
             if (batchSize != null) {
                 findOptions.batchSize(batchSize);
             }
+            if (allowDiskUse != null) {
```

Comment:
The find that we do on the collection we just out-ed to shouldn't need allowDiskUse just because the aggregate did.  There's no sort specified on this find, so allowDiskUse isn't relevant iiuc.  I think you can remove this change.

Reply:
Done